kotlin-generator/src/android-client.ts

```diff
@@ -0,0 +1,153 @@
+import { AstRoot } from "@sdkgen/parser";
+import { generateClass, generateEnum, generateErrorClass, generateJsonAddRepresentation, generateKotlinTypeName, mangle } from "./helpers";
+
+interface Options {}
+
+export function generateAndroidClientSource(ast: AstRoot, options: Options) {
+    let code = `import android.content.Context
+import android.util.Base64
+import com.google.gson.*
+import com.google.gson.reflect.TypeToken
+import io.sdkgen.runtime.SdkgenHttpClient
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.Dispatchers.IO
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.async
+import java.lang.reflect.Type
+import java.text.SimpleDateFormat
+import java.util.*
+
+inline fun <reified T> Gson.fromJson(json: String) =
+    fromJson<T>(json, object : TypeToken<T>() {}.type)
+
+inline fun <reified T> Gson.fromJson(json: JsonElement) =
+    fromJson<T>(json, object : TypeToken<T>() {}.type)
+
+@ExperimentalCoroutinesApi
+@Suppress("DeferredIsResult", "unused")
+class ApiClient(
+    baseUrl: String,
+    applicationContext: Context,
+    defaultTimeoutMillis: Long = 10000L
+) : SdkgenHttpClient(baseUrl, applicationContext, defaultTimeoutMillis) {`;
+
+    code += `
+    private class ByteArrayDeserializer : JsonDeserializer<ByteArray> {
+        @Throws(JsonParseException::class)
+        override fun deserialize(
+            element: JsonElement,
+            arg1: Type?,
+            arg2: JsonDeserializationContext?
+        ): ByteArray {
+            try {
+                return Base64.decode(element.asString, Base64.DEFAULT)
+            } catch (e: Exception) {
+                throw JsonIOException(e.message)
+            }
+        }
+    }
+
+    private class CalendarDeserializer : JsonDeserializer<Calendar> {
+        @Throws(JsonParseException::class)
+        override fun deserialize(
+            element: JsonElement,
+            arg1: Type?,
+            arg2: JsonDeserializationContext?
+        ): Calendar {
+            val date = element.asString
+            val formatter = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS", Locale.US)
+
+            formatter.timeZone = TimeZone.getTimeZone("UTC")
+
+            try {
+                return Calendar.getInstance().apply { time = formatter.parse(date)!! }
+            } catch (e: Exception) {
+                throw JsonIOException(e.message)
+            }
+        }
+    }
```

Comment:
These two classes should be moved to the runtime and imported from there

Reply:
I did this in a way to keep all the generated classes on the client instead of runtime. It was my understanding that this separation was part of the sdkgen concept, isn't it?

Reply:
But those classes aren't really generated. they are always the same.

Reply:
Yeah, but they handle serialization, which is a responsibility of the generator, should they be in another layer?

Reply:
I get you @danielpsantiago but I also think that they should be part of runtime because isn't actually generated code.

Reply:
All other runtimes have serialization code

Reply:
Ok, I don't agree, as we use a library to do the serialization on this target, but it's done! hahaha